docs/writing-rules/pattern-syntax.mdx

```diff
@@ -465,61 +465,46 @@ match:
 foo(1,2,3,1,2)
 ```
 
-### Displaying matched metavariable in rule message
+### Displaying matched metavariables in rule messages
 
-Display values of matched metavariables in rule messages. Add a metavariable to the rule message (for example `Found $X!`) and Semgrep replaces it with the value of the detected metavariable.
+Display values of matched metavariables in rule messages. Add a metavariable to the rule message (for example `Found $X`) and Semgrep replaces it with the value of the detected metavariable.
 
 For example, see the following rule message and part of a Semgrep rule (formula): 
 
 ```yaml
-- message: >-
-  Creating a buffer using $X
-- pattern: |
-   byte[] buf = new byte[$X];
+- message: Setting a password on $MODEL
+- pattern: $MODEL.set_password(…)
 ```
 
 Use the formula displayed above against the following code:
 
-```java
-byte[] buf = new byte[size];
+```python
+user.set_password(new_password)
 ```
 
 The resulting message is:
 
 ```
-Creating a buffer using size
+Setting a password on user
 ```
 
 Semgrep substitutes the $X with the value detected within the metavariable. In this case, the value was `size`.
 
-However, in some cases, the matched value of the metavariable is not the real value you were looking for. See the following formula and testing code:
-
-Part of a Semgrep rule (formula):
-
-```yaml
-- patterns:
-   - pattern: byte[] buf = new byte[$X];
-   - metavariable-comparison:
-        metavariable: $X
-        comparison: $X < 2048
-```
+To display matched metavariable in rule message:
 
-Testing code:
+1. Find the metavariable used in Semgrep rule. For example:
+    ```yaml
+    - pattern: $MODEL.set_password(…)
+    ```
+    This formula uses `$MODEL` as a metavariable.
+2. Insert the metavariable to rule message:
+    ```yaml
+    - message: Setting a password on $MODEL
+    ```
```

Comment:
Can we give them a way to try this out? Or at least indicate what the result would be?

Reply:
@emjin of course! Please, can you give me appropriate test code? 🙏

Reply:
Done))))